### PR TITLE
Replace FontDesc name with new FontFamily type

### PR DESCRIPTION
**Pull request by TatriX**
Currently default hardcoded typeface is "Arial", but this font is not
available by default everywhere.

This PR adds new FontFamily enum which can be used as an
alternative to the font name.

Also see #63

### CODE REVIEW

File: src/style/font/font_desc.rs [hunk 1]
```diff
@@ -45,20 +45,63 @@ impl FontTransform {
...
-    name: &'a str,
+    family: FontFamily<'a>,
     data: FontResult<FontDataInternal>,
     transform: FontTransform,
 }
 
+/// Describes font family
+#[derive(Clone, Copy)]
+pub enum FontFamily<'a> {
+    Sans,
```
Comment (38) [inline src/style/font/font_desc.rs]: Isn't "sans" same as "sans-serif" but "serif" is different ?
And for CSS it has "serif", "sans-serif" and "monospace". So it seems we should have `Serif` other than `Sans` https://developer.mozilla.org/en-US/docs/Web/CSS/font-family#%3Cgeneric-name%3E   

I am not familiar with fonts. But based on what I learned it seems to be this. I may be wrong on this.

**Comment by 38** [comment]
Hi @TatriX ,

Thanks for this PR. I have been thinking about this for a long time!  I think it looks really good. But my question is is "sans" and "sans-serif" the refers to the same thing but "serif" is different ? 

If so, the `FontFamily` enum may need some change. 

I am not very confident on this, I am not an expert on fonts. Please let me know if I was wrong.

Thanks again!

**Comment by TatriX** [comment]
I'm using [css font-family](https://developer.mozilla.org/en-US/docs/Web/CSS/font-family) notation.

**Comment by TatriX** [comment]
Sans Serif:

Serif:

> In typography and lettering, a sans-serif, sans serif, gothic, or simply sans letterform is one that does not have extending features called "serifs" at the end of strokes.

https://en.wikipedia.org/wiki/Sans-serif

File: src/style/font/font_desc.rs (hunk 1 above)
Comment (TatriX) [reply src/style/font/font_desc.rs]: Ah, right. I'm constantly misuse them ;)

**Comment by 38** [comment]
Thanks for this amazing work. I was planning this for a long time. 
Plus this should resolve issue #18 

really appreciate this work!

**Pull request merged**